system/Validation/Validation.php

```diff
@@ -364,6 +364,12 @@ protected function processRules(string $field, string $label = null, $value, $ru
 	 */
 	public function withRequest(RequestInterface $request): ValidationInterface
 	{
+		if ($request->hasHeader('Content-Type') && $request->getHeader('Content-Type')->getValue() === 'application/json')
```

Comment:
I see this code or similar a lot. Maybe it is time for `Request::isJSON()`? Maybe even on `Message`? @caswell-wc you appear to work with JSON a lot in the HTTP/API layer - what do you think?

Reply:
I have been thinking about suggesting something like that. I think it's a good idea. I have some other ideas on how we could improve the request and having that method would be helpful.

Reply:
I went ahead and created that method and pushed it into this PR.

Reply:
I like it! Since we already have `getJSON()` could you please make the capitalization consistent? `isJSON()`

Reply:
That's fixed